core/repository/sparql/src/test/java/org/eclipse/rdf4j/repository/sparql/SPARQLConnectionTest.java

```diff
@@ -0,0 +1,38 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Eclipse RDF4J contributors.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Distribution License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *******************************************************************************/
+package org.eclipse.rdf4j.repository.sparql;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+import org.eclipse.rdf4j.http.client.SPARQLProtocolSession;
+import org.eclipse.rdf4j.rio.ParserConfig;
+import org.junit.Before;
+import org.junit.Test;
+
+public class SPARQLConnectionTest {
+
+	private SPARQLConnection subject;
+	private SPARQLProtocolSession client;
+
+	@Before
+	public void setUp() throws Exception {
+		client = mock(SPARQLProtocolSession.class);
+		subject = new SPARQLConnection(null, client);
+	}
+
+	@Test
+	public void setParserConfigPassesToProtocolSession() throws Exception {
+		ParserConfig config = new ParserConfig();
+
+		subject.setParserConfig(config);
+		verify(client, times(1)).setParserConfig(config);
```

Comment:
Haven’t seen this part of mockito. This tests that the setParserConfig method has been called exactly once with `config`?

Reply:
That's exactly right! The client is a mock, and we use verify to check that a particular method with a particular argument has been called on that mock.